File: 프로그래머스/3/43164. 여행경로/여행경로.py
def solution(tickets):

    routes = dict()

    for (start, end) in tickets:
        routes[start] = routes.get(start, []) + [end]  
        
   
    for r in routes.keys():
        routes[r].sort(reverse=True)

    way = ["ICN"]
    path = []

    while way:
        top = way[-1]
        if top not in routes or len(routes[top]) == 0:
            path.append(way.pop())
        else:
            way.append(routes[top][-1])
            routes[top] = routes[top][:-1]
    
    answer = path[::-1]
    return answer

File: 프로그래머스/3/43164. 여행경로/여행경로.js
var solution = function (tickets) {
  const graph = new Map();

  // 그래프 생성
  for (const [src, dst] of tickets) {
    if (!graph.has(src)) graph.set(src, []);
    graph.get(src).push(dst);
  }
  console.log(graph);
  // 목적지 리스트 정렬
  for (const [key, value] of graph) {
    value.sort().reverse(); // 역순 정렬하여 pop() 사용 가능
  }
  console.log(graph);

  const itinerary = [];

  function dfs(airport) {
    while (graph.has(airport) && graph.get(airport).length > 0) {
      dfs(graph.get(airport).pop());
    }
    itinerary.push(airport);
  }

  dfs("ICN");

  return itinerary.reverse();
};